MekHQ/src/mekhq/campaign/finances/Loan.java

```diff
@@ -222,43 +224,44 @@ public void paidLoan() {
-     * and campaign-specific conditions such as the Gray Monday event.
+     * Computes and returns a base loan object based on the player's rating, the current date, and campaign-specific
+     * conditions such as the Gray Monday event.
      *
      * <p>This method determines the loan terms that the player is eligible for based on their
-     * performance rating and whether the game is simulating the Gray Monday event. If the
-     * Gray Monday event is active, a special predatory loan with significantly higher interest
-     * rates and penalties is offered. Otherwise, the loan terms are progressively better
-     * as the player's rating improves.</p>
+     * performance rating and whether the game is simulating the Gray Monday event. If the Gray Monday event is active,
+     * a special predatory loan with significantly higher interest rates and penalties is offered. Otherwise, the loan
+     * terms are progressively better as the player's rating improves.</p>
      *
      * <p>The returned {@link Loan} object contains all relevant terms such as the principal,
      * interest rate, repayment duration, financial term, and associated penalty.</p>
      *
-     * @param rating             The player's performance rating as an integer. Defaults to higher
-     *                           loan penalties and stricter terms for lower ratings.
-     * @param simulateGrayMonday A {@code boolean} flag that indicates whether the Gray Monday
-     *                           event is active, which impacts loan terms significantly.
-     * @param date               The current in-game date as a {@link LocalDate} object, used
-     *                           to determine if Gray Monday conditions apply.
+     * @param rating             The player's performance rating as an integer. Defaults to higher loan penalties and
+     *                           stricter terms for lower ratings.
+     * @param simulateGrayMonday A {@code boolean} flag that indicates whether the Gray Monday event is active, which
+     *                           impacts loan terms significantly.
+     * @param campaign           The current campaign object, used for LocalDate to determine if Gray Monday conditions
+     *                           apply and for fetching faction.
      *
-     * @return A {@link Loan} object representing the player's base loan terms based on
-     *         their rating and event conditions.
+     * @return A {@link Loan} object representing the player's base loan terms based on their rating and event
+     *       conditions.
      */
-    public static Loan getBaseLoan(final int rating, boolean simulateGrayMonday, final LocalDate date) {
+    public static Loan getBaseLoan(final int rating, boolean simulateGrayMonday, final Campaign campaign) {
         // we are going to treat the score from StellarOps the same as dragoons score
-        // TODO: pirates and government forces
 
+        LocalDate date = campaign.getLocalDate();
```

Comment:
If you need the date from `Campaign#getLocalDate()` you pass the value from it, not the campaign.